includes/admin/class-wc-admin-setup-wizard.php

```diff
@@ -1474,19 +1560,24 @@ class="jetpack-logo"
 	public function wc_setup_activate_save() {
 		check_admin_referer( 'wc-setup' );
 
-		// Determine if we need to install Jetpack synchronously here.
+		// Clean up temporary Jetpack queued install option.
+		// This happens after the connection button is clicked
+		// and we waited for the pending install to finish.
+		delete_option( 'woocommerce_setup_queued_jetpack_install' );
 
+		$result = $this->install_and_activate_jetpack_sync();
 
-		if ( ! class_exists( 'Jetpack' ) ) {
-			wp_redirect( esc_url_raw( add_query_arg( 'activate_error', 'install' ) ) );
+		if ( true !== $result ) {
+			wp_redirect( esc_url_raw( add_query_arg( 'activate_error', $result ) ) );
 			exit;
 		}
 
 		Jetpack::maybe_set_version_option();
 		$registered = Jetpack::try_registration();
 
 		if ( is_wp_error( $registered ) ) {
-			wp_redirect( esc_url_raw( add_query_arg( 'activate_error', 'register' ) ) );
+			$error_code = array_key_exists( $registered->get_error_code(), $this->get_all_activate_errors() ) ? $registered->get_error_code() : 'register';
```

Comment:
Let's avoid potentially calling `$registered->get_error_code()` twice here with a variable.

Reply:
Ok. Updated & rebased, new version is at 518220e5b07e82a0352a182a7f7c81d1acd3760e